### PR TITLE
Introduction of the possibility to start devbuilds manually and to publish the artifacts to pypi/testpypi

### PR DESCRIPTION
**Pull request by andife**
### Description
* Introduction of the possibility to start devbuilds manually and to publish the artifacts to pypi/testpypi
* Harmonization of code and wording in various places

### Motivation and Context
Currently there is only one attempt per week to build a devbuild, currently called onnx-weekly via the pipeline, if there are now errors in the pipeline or at the interfaces, you only have one attempt per week...

An error for which this functionality would have been helpful
https://github.com/onnx/onnx/issues/6499

### CODE REVIEW

File: .github/workflows/release_linux_aarch64.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
       os:
         required: true
         type: string
-
+
```
Comment (justinchuby) [inline .github/workflows/release_linux_aarch64.yml:13]:
```suggestion

```

File: .github/workflows/release_sdist.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ on:  # Specifies the event triggering the workflow
       os:
         required: true
         type: string
-
+
```
Comment (justinchuby) [inline .github/workflows/release_sdist.yml:13]:
```suggestion

```

File: .github/workflows/release_win.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ on:  # Specifies the event triggering the workflow
       os:
         required: true
         type: string
-
+
```
Comment (justinchuby) [inline .github/workflows/release_win.yml:13]:
```suggestion

```

**Review by justinchuby — APPROVED** [review]
LGTM overall, thanks! Does this mean though we can have dev builds that are not weekly?

**Comment by andife** [comment]
> LGTM overall, thanks! Does this mean though we can have dev builds that are not weekly?

yes, at least that would be the idea of the workflow

image


---
I think we also have to think about the names again... I mean what distinguishes the Weekly Build / DevBuild from the “normal” build? Nothing but the date in the name?

File: .github/workflows/create_release.yml [hunk 4]
```diff
@@ -5,141 +5,157 @@
...
     - cron:  '00 00 * * MON'
   push:
     branches: [main, rel-*]
   pull_request:
     branches: [main, rel-*]
-    types: 
+    types:
       - labeled  # Trigger when a label is added to a PR, more information: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request
-  workflow_dispatch:    
+  workflow_dispatch:
```
Comment (justinchuby) [inline .github/workflows/create_release.yml:16]: We need to ensure it can only be triggered on the main branch

Comment (andife) [reply .github/workflows/create_release.yml:16]: I think we need to make sure that only main-branch wheels can be published

Comment (andife) [reply .github/workflows/create_release.yml:16]: We check for that: (github.ref == 'refs/heads/main')

Comment (justinchuby) [reply .github/workflows/create_release.yml:16]: Thanks!

**Pull request merged**